service/scaladsl/client/src/main/scala/com/lightbend/lagom/scaladsl/client/ServiceClient.scala

```diff
@@ -141,31 +144,134 @@ trait LagomServiceClientComponents extends TopicFactoryProvider { self: LagomCon
  * It is important to invoke [[#stop]] when the application is no longer needed, as this will trigger the shutdown
  * of all thread and connection pools.
  */
+@deprecated(message = "Use StandaloneLagomClientFactory instead", since = "1.5.0")
 abstract class LagomClientApplication(
   clientName:  String,
   classLoader: ClassLoader = classOf[LagomClientApplication].getClassLoader
+) extends StandaloneLagomClientFactory(clientName, classLoader)
```

Comment:
This was a bit confusing. The java dsl has a `LagomClientFactory`, while the Scala one is a `LagomClientApplication`. 

This is not an application, it is still a factory. One has to build a factory and use it to build clients. 

I also need a way to differentiate from a factory that manages its own actor system and materializer from one that has them injected. Therefore, I'm suggesting to deprecate the name `LagomClientApplication` and introduce a `StandaloneLagomClientFactory` and a `LagomClientFactory`. 

Here we solve a few things: 
1. we align with the java dsl
2. we an implementation that is clear on its purpose, is a standalone
3. we have the abstract `LagomClientFactory` that gives the user to define the `ActorSystem` and `Materializer` they way they want. The may inject them, they can choose to manage it internally and load whatever config they want. It's up to them.

Reply:
That's great. Simpler and easy to explain. +1